### PR TITLE
Feature/us7 password recovery

### PR DESCRIPTION
**Pull request by BlessedLongsword**
This should meet all acceptance criteria for US7. Broadly

User clicks on forgot password in login
User types their email (Validations of format  and existence of the email in the database are done)
The email is sent (this is done through a backend request)
The user is redirected to the password update page
The user needs to input the verification code sent through email and the updated passwords (verification of fields is done)
If everything is correct, when entering there is a confirmation popup and the user is sent to login.

Let me know if you need any change.

### CODE REVIEW

**Comment by BlessedLongsword** [comment]
As of now it is necessary to be in the branch of the pull request: https://github.com/UB-ES-2023-A2/kasula-backend/pull/53
I'll warn @andreu-vall and everyone that everything I've done is intended to work locally. Basically, you will have to change all request url to the production environment one. in the FUTURE

File: frontend/src/components/PasswordChange.jsx [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+                    </InputGroup>
+                  </Form.Group>
+                  <Form.Group className="mb-5" controlId="formConfirmPassword">
+                    <Form.Label>Confirm Password</Form.Label>
+                    <InputGroup>
+                      <Form.Control
+                        className={
+                          confirmPasswordValidated
+                            ? passwordsMatch
+                              ? "is-valid"
```
Comment (ivanmansilla) [inline frontend/src/components/PasswordChange.jsx:288]: I think it is a good idea to have several steps to ensure a good password, such as confirming the password.

File: frontend/src/components/PasswordRecovery.jsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+      setEmailValid(false);
+    } else if (email.search(/^[^\s@]+@[^\s@]+\.[^\s@]+$/) < 0) {
+      setEmailErrorMessage("Please enter a valid email address");
+      setEmailValid(false);
+    } else {
+      setEmailErrorMessage("");
+      setEmailValid(true);
+    }
+    setEmailValidated(true);
+  };
```
Comment (ivanmansilla) [inline frontend/src/components/PasswordRecovery.jsx:37]: Complies with the Acceptance Criteria that you enter a valid email address.

File: frontend/src/components/PasswordRecovery.jsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        const data = await response.json();
+        setEmailValid(false);
+        if (data && data.detail) {
+          setEmailErrorMessage(data.detail);
+        } else {
+          setEmailErrorMessage("Could not send email. Please try again later.");
+        }
+      } else {
+        navigate("/passwordrecovery/set", {
+          state: { email: email },
```
Comment (ivanmansilla) [inline frontend/src/components/PasswordRecovery.jsx:98]: Complies with the Acceptance Criteria that send a email to password recovery

File: frontend/src/components/PasswordChange.jsx [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+          setUpdateConfirmationMessage(
+            "Could not check verification code. Please try again later."
+          );
+          setShowUpdateConfirmation(true);
+        }
+      } else {
+        const data = await response.json();
+        setVerificationCodeValid(data.status);
+        if (!data.status) {
+          setVerificationCodeErrorMessage("Verification code is incorrect.");
```
Comment (ivanmansilla) [inline frontend/src/components/PasswordChange.jsx:153]: Good security measures, making it necessary to enter a verification code to change the password.

File: frontend/src/components/PasswordChange.jsx [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+              <p>{updateConfirmationMessage}</p>
+            </Col>
+          </Row>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button
+            variant={updateSuccess ? "success" : "secondary"}
+            onClick={() => navigate("/login")}
+          >
+            {updateSuccess ? "Go to login" : "Close"}
```
Comment (ivanmansilla) [inline frontend/src/components/PasswordChange.jsx:370]: Fulfills the missing aceptance criteria of updating the password and redirecting to the login.

**Review by ivanmansilla — APPROVED** [review]

**Pull request merged**